src/main/kotlin/no/nav/helsearbeidsgiver/bakgrunnsjobb/LeaderElectedBakgrunnsjobbService.kt

```diff
@@ -0,0 +1,104 @@
+package no.nav.helsearbeidsgiver.bakgrunnsjobb
+
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.engine.apache5.Apache5
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.request.get
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.Serializable
+import no.nav.hag.utils.bakgrunnsjobb.BakgrunnsjobbProsesserer
+import no.nav.hag.utils.bakgrunnsjobb.BakgrunnsjobbRepository
+import no.nav.hag.utils.bakgrunnsjobb.BakgrunnsjobbService
+import no.nav.hag.utils.bakgrunnsjobb.RecurringJob
+import no.nav.helsearbeidsgiver.Env.getPropertyOrNull
+import no.nav.helsearbeidsgiver.utils.json.jsonConfig
+import no.nav.helsearbeidsgiver.utils.log.sikkerLogger
+import java.net.InetAddress
+import java.time.LocalDateTime
+
+class LeaderElectedBakgrunnsjobbService(
+    delayMillis: Long = 30 * 1000L,
+    coroutineScope: CoroutineScope = CoroutineScope(Dispatchers.IO),
+    bakgrunnsjobbRepository: BakgrunnsjobbRepository,
+) : RecurringJob(coroutineScope, delayMillis) {
+    val bakgrunnsjobbService = BakgrunnsjobbService(bakgrunnsjobbRepository)
+
+    inline fun <reified T : BakgrunnsjobbProsesserer> opprettJobb(
+        kjoeretid: LocalDateTime = LocalDateTime.now(),
+        forsoek: Int = 0,
+        maksAntallForsoek: Int = 3,
+        data: String,
+    ) {
+        bakgrunnsjobbService.opprettJobb<T>(
+            kjoeretid = kjoeretid,
+            forsoek = forsoek,
+            maksAntallForsoek = maksAntallForsoek,
+            data = data,
+        )
+    }
+
+    fun registrer(prosesserer: BakgrunnsjobbProsesserer) {
+        bakgrunnsjobbService.registrer(prosesserer)
+    }
+
+    override fun doJob() {
+        if (isElectedLeader()) {
+            do {
+                val wasEmpty =
+                    bakgrunnsjobbService
+                        .finnVentende()
+                        .also { logger.debug("Bakgrunnsjobb: Fant ${it.size} bakgrunnsjobber å kjøre på ${getElectedLeaderId()}") }
+                        .onEach { bakgrunnsjobbService.prosesser(it) }
+                        .isEmpty()
+            } while (!wasEmpty)
```

Comment:
Litt smak og behag her, men hva tror du om å bruke `isNotEmpty()` og endre variabelen i while til `fantVentende` sånn her?

```suggestion
                val fantVentende =
                    bakgrunnsjobbService
                        .finnVentende()
                        .also { logger.debug("Bakgrunnsjobb: Fant ${it.size} bakgrunnsjobber å kjøre på ${getElectedLeaderId()}") }
                        .onEach { bakgrunnsjobbService.prosesser(it) }
                        .isNotEmpty()
            } while (fantVentende)
```

Reply:
Denne koden er kopiert fra bakgrunnsjobb-repoen, og jeg ville holde den lik. Jeg er helt enig med deg – det er bedre å bruke sanne betingelser. Kanskje vi kan endre det når vi flytter evt. koden ditt.

Reply:
Skjønner 👌